drivers/media/i2c/imx708.c

```diff
@@ -1098,6 +1095,23 @@ static int imx708_set_ctrl(struct v4l2_ctrl *ctrl)
 	if (ctrl->id == V4L2_CID_VBLANK)
 		imx708_adjust_exposure_range(imx708, ctrl);
 
+	/*
+	 * The WIDE_DYNAMIC_RANGE control can also be applied immediately as it
+	 * doesn't set any registers, though we shouldn't do anything if the
+	 * control value already matches the mode.
+	 */
+	if (ctrl->id == V4L2_CID_WIDE_DYNAMIC_RANGE &&
+	    imx708->mode && imx708->mode->hdr != ctrl->val) {
```

Comment:
I thought we could get the old value from the control handler rather than having to cache it ourselves, but that may well be my memory. ctrl->val is certainly the new value, and I don't know where else it would be kept.

Isn't imx708->mode meant to be protected by the mutex? Or as we're using the same mutex for both driver and ctrl_handler, perhaps that is all safe anyway?

Reply:
imx708->mode seemed to be fair game before I came along, so I'll defer to the camera gods on that one! I'll put in the switch statement though...